src/plugins/index_pattern_management/public/components/create_index_pattern_wizard/components/step_index_pattern/components/header/header.tsx

```diff
@@ -6,139 +6,247 @@
+  EuiComboBox,
+  EuiComboBoxOptionOption,
 } from '@elastic/eui';
 
 import { i18n } from '@kbn/i18n';
 import { FormattedMessage } from '@kbn/i18n/react';
+import { containsIllegalCharacters } from '../../../../lib';
+import { indexPatterns } from '../../../../../../../../data/public';
 
 interface HeaderProps {
-  isInputInvalid: boolean;
-  errors: any;
   characterList: string;
-  query: string;
-  onQueryChanged: (e: React.ChangeEvent<HTMLInputElement>) => void;
-  goToNextStep: (query: string) => void;
+  checkIndices?: string[];
+  goToNextStep: (selectedPatterns: string[]) => void;
+  isIncludingSystemIndices: boolean;
   isNextStepDisabled: boolean;
-  showSystemIndices?: boolean;
   onChangeIncludingSystemIndices: (event: EuiSwitchEvent) => void;
-  isIncludingSystemIndices: boolean;
+  onQueryChanged: (patterns: string[]) => void;
+  onTitleChanged: (e: React.ChangeEvent<HTMLInputElement>) => void;
+  patternError: boolean;
+  selectedPatterns: string[];
+  showSystemIndices?: boolean;
+  title: string;
+  titleError: boolean;
 }
 
 export const Header: React.FC<HeaderProps> = ({
-  isInputInvalid,
-  errors,
   characterList,
-  query,
-  onQueryChanged,
+  checkIndices,
   goToNextStep,
+  isIncludingSystemIndices,
   isNextStepDisabled,
-  showSystemIndices = false,
   onChangeIncludingSystemIndices,
-  isIncludingSystemIndices,
+  onQueryChanged,
+  onTitleChanged,
+  patternError,
+  selectedPatterns,
+  showSystemIndices = false,
+  title,
+  titleError,
   ...rest
-}) => (
-  <div {...rest}>
-    <EuiTitle size="s">
-      <h2>
-        <FormattedMessage
-          id="indexPatternManagement.createIndexPattern.stepHeader"
-          defaultMessage="Step 1 of 2: Define an index pattern"
-        />
-      </h2>
-    </EuiTitle>
-    <EuiSpacer size="m" />
-    <EuiFlexGroup>
-      <EuiFlexItem>
-        <EuiForm isInvalid={isInputInvalid}>
-          <EuiFormRow
-            fullWidth
-            label={
-              <FormattedMessage
-                id="indexPatternManagement.createIndexPattern.step.indexPatternLabel"
-                defaultMessage="Index pattern name"
-              />
-            }
-            isInvalid={isInputInvalid}
-            error={errors}
-            helpText={
-              <>
+}) => {
+  const [isPatternInvalid, setIsPatternInvalid] = useState(patternError);
+  const selectedOptions: Array<EuiComboBoxOptionOption<string>> = useMemo(
+    () => selectedPatterns.map((label) => ({ label })),
+    [selectedPatterns]
+  );
+  const onChange = useCallback(
+    (patterns: Array<EuiComboBoxOptionOption<string>>) => {
+      const patternList = patterns.map(({ label }) => label);
+      onQueryChanged(patternList);
+    },
+    [onQueryChanged]
+  );
+  const onCreateOption = useCallback(
+    (searchValue) => {
+      if (containsIllegalCharacters(searchValue, indexPatterns.ILLEGAL_CHARACTERS)) {
+        return false;
+      }
+      onQueryChanged([...selectedPatterns, searchValue]);
```

Comment:
not sure that this code works correctly 

if I input `index-a,index-b,index-c`. It should be as least splitted by comma and added as a three separate indexes.

https://user-images.githubusercontent.com/20072247/106265477-2302aa80-6238-11eb-8054-fa794b1c33f7.mov


**to discuss**: some time ago we tried to solve the same problem on `TSVB` plugin. From my point of view here instead of `EuiComboBox` we should use `EuiSuggest` component.  See #82696